### DISCUSSION TITLE
Pluto and PackageCompiler

### DISCUSSION THREAD

**Discussion by gianmariomanca** [fonsp/Pluto.jl#1415] [Questions about using Pluto]
Does Pluto with its own new package management system play well with PackageCompiler?
If I have a sysimage for fast plotting and I start julia with
`julia -t 8 --sysimage sys_plots.so`
should I expect the pluto session to use the sysimage and so have faster time to first plot?

**Comment by ctrekker** [comment]
I'm not sure the sysimage will get loaded by the Pluto workers even if you start Julia like that. When you start Pluto though you should be able to pass a `sysimage` keyword argument into `Pluto.run`. For example,

```julia
import Pluto
Pluto.run(; sysimage="sys_plots.so")
```

Pluto has several other options that get sent to newly launched Julia processes grouped under [`CompilerOptions`](https://github.com/fonsp/Pluto.jl/blob/ac67ba8bd40d91bc101e350c6934791b94b1cb45/src/Configuration.jl#L114)

**Reply by gianmariomanca** [reply]
What happens if the version of Plots hardcoded in the sysimage is different from the version in the notebook manifest? Would Pluto still be happy and go with the sysimage?

**Reply by manuelbb-upb** [reply]
I just stumbled upon this very issue. I have compiled a sysimage with Plots@1.32 in it, but the built-in package manager does not seem to honor the sysimage when doing `using Plots`. Then, the newest version (1.36) is installed and the TTFP is long.
With a cell à la
```julia
begin
    import Pkg
    Pkg.add("Plots")
end
```
the version is resolved correctly and TTFP is short.

**Edit**: This was in a pre-existing notebook file. I just created a new notebook and there was a `Pkg` cell by default and an explanatory comment :smile:
